From 6914bb60ac23b4b03520202046ae0d4ab1e734ad Mon Sep 17 00:00:00 2001
From: Robert Helmer <rhelmer@rhelmer.org>
Date: Tue, 24 Feb 2015 07:44:03 -0800
Subject: [PATCH] add basic systemd service option

---
 lib/fpm/package/rpm.rb | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/lib/fpm/package/rpm.rb b/lib/fpm/package/rpm.rb
index 8310d51..48ffc73 100644
--- a/lib/fpm/package/rpm.rb
+++ b/lib/fpm/package/rpm.rb
@@ -104,6 +104,11 @@ class FPM::Package::RPM < FPM::Package
     next File.expand_path(file)
   end
 
+  option "--service", "FILEPATH", "Add FILEPATH as a systemd service",
+       :multivalued => true do |file|
+    next File.expand_path(file)
+  end
+
   rpmbuild_filter_from_provides = []
   option "--filter-from-provides", "REGEX",
     "Set %filter_from_provides to the supplied REGEX." do |filter_from_provides|
@@ -471,6 +476,14 @@ def output(output_path)
       File.chmod(0755, dest_init)
     end
 
+    # add service script if present
+    (attributes[:rpm_service_list] or []).each do |service|
+      name = File.basename(service)
+      dest_service = File.join(staging_path, "usr/lib/systemd/system/#{name}")
+      FileUtils.mkdir_p(File.dirname(dest_service))
+      FileUtils.cp service, dest_service
+    end
+
     (attributes[:rpm_rpmbuild_define] or []).each do |define|
       args += ["--define", define]
     end
